rest/api.go

```diff
@@ -156,35 +156,31 @@ func (h *handler) handleFlush() error {
 }
 
 func (h *handler) handleGetResync() error {
-	h.writeJSON(h.db.ResyncManager.GetStatus())
+	h.writeRawJSON(h.db.ResyncManager.GetStatus())
 	return nil
 }
 
 func (h *handler) handlePostResync() error {
 	action := h.getQuery("action")
 	regenerateSequences, _ := h.getOptBoolQuery("regenerate_sequences", false)
 
-	if action != "" && action != db.ResyncActionStart && action != db.ResyncActionStop {
+	if action != "" && action != string(db.BackgroundProcessActionStart) && action != string(db.BackgroundProcessActionStop) {
 		return base.HTTPErrorf(http.StatusBadRequest, "Unknown parameter for 'action'. Must be start or stop")
 	}
 
 	if action == "" {
-		action = db.ResyncActionStart
+		action = string(db.BackgroundProcessActionStart)
 	}
 
-	if action == db.ResyncActionStart {
+	if action == string(db.BackgroundProcessActionStart) {
 		if atomic.CompareAndSwapUint32(&h.db.State, db.DBOffline, db.DBResyncing) {
-			h.db.ResyncManager.SetRunStatus(db.ResyncStateRunning)
-			h.writeJSON(h.db.ResyncManager.GetStatus())
-			go func() {
-				defer atomic.CompareAndSwapUint32(&h.db.State, db.DBResyncing, db.DBOffline)
-				defer h.db.ResyncManager.SetRunStatus(db.ResyncStateStopped)
-				_, err := h.db.UpdateAllDocChannels(regenerateSequences)
-				if err != nil {
-					base.Errorf("Error occurred running resync operation: %v", err)
-					h.db.ResyncManager.SetError(err)
-				}
-			}()
+			h.db.ResyncManager.ResetStatus()
+			h.db.ResyncManager.SetRunState(db.BackgroundProcessStateRunning)
```

Comment:
I don't think we should require external actors to set the run state - it would be preferablce for BackgroundManager to manage that internally in response to the Start/Stop actions.

In this particular case, it looks to me like we could set the status to 'running' inside 'Start' prior to starting the async goroutine, and perform the GetStatus/writeRawJSON after Start returns.